django_auth_adfs/backend.py

```diff
@@ -222,8 +222,8 @@ def update_user_groups(self, user, claims):
                              settings.GROUPS_CLAIM)
                 claim_groups = []
             if sorted(claim_groups) != sorted(django_groups):
-                existing_groups = list(Group.objects.filter(name__in=claim_groups).iterator())
-                existing_group_names = frozenset(group.name for group in existing_groups)
+                existing_groups = Group.objects.filter(name__in=claim_groups)
+                existing_group_names = frozenset(existing_groups.values_list("name", flat=True))
```

Comment:
This `values_list` call will generate a SQL query and the lazy`existing_groups` is casted to list later (one query), the new code adds a new SQL query.